src/main/java/angel/Parser.java

```diff
@@ -0,0 +1,265 @@
+        }
+        String[] eventDetails = parts[1].split(" /from ");
+        String[] times = eventDetails[1].split(" /to ");
+        LocalDateTime fromDate = parseDate(times[0]);
+        LocalDateTime toDate = parseDate(times[1]);
+        return new Command("event", new EventDetail(eventDetails[0], fromDate, toDate));
+    }
+
+    /**
+     * Parses the command for operations requiring a task index (e.g., mark, unmark, delete).
+     *
+     * @param parts The parts of the user input split by space.
+     * @param command The command type (e.g., mark, unmark, delete).
+     * @return A Command object containing the command type and task index.
+     * @throws InvalidCommandException If the task index is missing or not a valid number.
+     */
+    private static Command parseIndexCommand(String[] parts, String command) throws InvalidCommandException {
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new InvalidCommandException("Please specify the task number to " + command + ".");
+        }
+        try {
+            int taskIndex = Integer.parseInt(parts[1]) - 1;
+            return new Command(command, taskIndex);
+        } catch (NumberFormatException e) {
+            throw new InvalidCommandException("Invalid task number for '" + command + "'. Please enter a valid number after '" + command + "'.");
+        }
+    }
+
+    /**
+     * Parses a date/time string into a LocalDateTime object.
+     *
+     * @param dateTimeString The date/time string in 'yyyy-MM-dd HHmm' format.
+     * @return A LocalDateTime object representing the parsed date/time.
+     * @throws InvalidCommandException If the date/time format is invalid.
+     */
+    private static LocalDateTime parseDate(String dateTimeString) throws InvalidCommandException {
+        try {
+            return LocalDateTime.parse(dateTimeString, INPUT_FORMAT);
+        } catch (DateTimeParseException e) {
+            throw new InvalidCommandException("Invalid date/time format. Please use 'yyyy-MM-dd HHmm'.");
+        }
+    }
+
+    /**
+     * Represents a command parsed from user input.
+     */
+    public static class Command {
+        private String type;
+        private Object details;
+
+        /**
+         * Constructs a Command object.
+         *
+         * @param type The type of the command.
+         * @param details The details associated with the command.
+         */
+        public Command(String type, Object details) {
+            this.type = type;
+            this.details = details;
+        }
+
+        /**
+         * Gets the type of the command.
+         *
+         * @return The command type.
+         */
+        public String getType() {
+            return type;
+        }
+
+        /**
+         * Gets the details of the command.
+         *
+         * @return The command details.
+         */
```

Comment:
Perhaps the object type that is returned could be specified in the description of the function.

Reply:
Okay got it!